src/pages/api/checkAvail.tsx

```diff
@@ -5,35 +5,53 @@ import {
   makeRes,
   getLanguageInfo,
 } from "../../lib/backend/database-utils";
+import { addDays, getNextWeekday } from "../../lib/frontend/utils";
 
 //this function will check the availability
 const handler = async (req: NextApiRequest, res: NextApiResponse) => {
-  console.log("in api with ", req.body);
+
   if (req.method == "POST") {
-    //first get the information for the requested date and language to see number of seats taken already
-    var data = getDateInfo(req.body.date, req.body.language).then(
-      (returned) => {
-        console.log("passed", req.body.date, req.body.language);
-        //get the max available number of seats possible for a given language
-        getLanguageInfo(req.body.language).then((languageRes) => {
-          if (languageRes != undefined) {
-            var max_seats =
-              languageRes[0].tablesOf8 * 8 +
-              languageRes[0].tablesOf6 * 6 -
-              languageRes[0].reserved_seats;
-            console.log(max_seats);
-            var takenSeats = returned.length;
-            console.log("len", takenSeats);
-            var availible = max_seats - takenSeats;
-            res.status(200).json({ data: availible });
-          } else {
-            res.status(200).json({ data: 0 });
-          }
-        });
-        console.log(returned);
-        console.log(req.body.date);
+    
+    console.log("in getAvail API call. the passed data is: ", req.body.data, ", and the language is:", req.body.langauge)
+
+    let returnDict: { [key: string]: string } = {};
+    //TODO - make today in this time format (T04 is important)
+    // const today = new Date("2022-10-18T04:00:00.000Z");//2022-10-18T04:00:00.000Z
+    const today = new Date().toISOString().split("T")[0] + "T05:00:00.000Z"
+    console.log("today's date ", today)
+
+    //first see how many max seats we have for that language
+    var languageRes = await getLanguageInfo(req.body.language)
+    if (await languageRes != undefined) {
+      var max_seats =
+        await languageRes[0].tablesOf8 * 8 +
+        await languageRes[0].tablesOf6 * 6 -
+        await languageRes[0].reserved_seats;
+        console.log("max seats: ", max_seats);
+
+      //check the availability for each date in the next 30 days
+      for (var i = 0; i < 30; i++) {
```

Comment:
Huge no-no here.  Looped DB calls should almost never be used and is very bad for performance.  Look into the "N+1 query problem".  I've refactored this code in the reservations endpoint to work with a single query.  Take a look.